client/internal/providerqueryer/queryer.go

```diff
@@ -0,0 +1,114 @@
+package providerqueryer
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	"golang.org/x/sync/semaphore"
+)
+
+const (
+	maxConcurrency      = 6
```

Comment:
why not a power of two ?

Reply:
This is what was used before so I just kept it the same to constrain the scope, so too many things aren't changing at once.

Reply:
I hate it